rust_src/src/data.rs

```diff
@@ -789,4 +790,36 @@ pub fn logcount(value: EmacsInt) -> i32 {
     unsafe { rust_count_one_bits(value as usize) }
 }
 
+/// Parse STRING as a decimal number and return the number.
+/// Ignore leading spaces and tabs, and all trailing chars.  Return 0 if
+/// STRING cannot be parsed as an integer or floating point number.
+///
+/// If BASE, interpret STRING as a number in that base.  If BASE isn't
+/// present, base 10 is used.  BASE must be between 2 and 16 (inclusive).
+/// If the base used is not 10, STRING is always parsed as an integer.
+#[lisp_fn(min = "1", name = "string-to-number", c_name = "string_to_number")]
+pub fn string_to_number_lisp(mut string: LispStringRef, base: Option<EmacsInt>) -> LispObject {
+    let b = match base {
+        None => 10,
+        Some(n) => {
+            if !(n >= 2 && n <= 16) {
```

Comment:
```suggestion
            if (n < 2 || n > 16) {
```

DeMorgan's FTW!

Reply:
You beat me to it by mere minutes.

Reply:
Damn. @shaleh I intended to keep this in mind after you told me this in your PR =)

Reply:
I really like this new suggestions mechanism. I have left feedback for GitHub that is needs to support multiline initial selections. I can only replace a single line of original text.

Reply:
Can you apply this @brotzeit ?